### ISSUE TITLE
question about normalizing to zero mean and unary variance

### ISSUE THREAD

**Issue by shirley0106** [Kamnitsask/deepmedic#56]
Hi,

My 3D CT data all have different dimensions. When you say normalizing to zero mean and unary variance you mean normalizing over all the training data or each data? And If normalizing over all training data how should I do it since all the dimensions are different?
Thanks!

**Comment by Kamnitsask** [comment]
Hi,

Normalizing to 0-mean 1-std does not have to do with different dimensions of data.

(a) Normalizing the intensities to 0-mean 1-std is suggested for various reasons. One of them is that the meta-parameters' values of the default config were chosen when working on such normalized data. It is possible that they wont work well for data with intensities in other scales and would require reconfiguration. CNNs in general behave better when input follows the 0mean 1std convention, though it can totally work in other input ranges (eg data rescaled to [0,1] or [-1,+1]), but reconfig of some parameters such as learning rate might be needed by you. I cannot be advicing on this.

The suggested normalization to 0mean 1std is on a "per image" basis. ie, just subtract from each image each own mean, and then divide with its std. (if constrained ROI can be derived, eg a brainmask, calculate mean and std within the roi only is adviced based on my experience).

(b) Data dimensions: data of a particular subject should have same dimensions. ie, the channel, roi and ground truth label images of a particular subject should all have same dimensions.

The images of all subjects of a database do NOT need to have the same dimensions. ie, same dimensions for the images of each subject individual "yes", across the database "not needed".

What IS adviced (strongly) across the database, is that all images have the same voxel size. A voxel in any image of the database should correspond to the same physical dimensions (eg 1mm x 1mm x 1mm). This is so that the kernels of the CNN will correspond to the same physical dimension when processing any of the subjects). This can be done via "resampling", an operation available in various softwares/libraries (see IRTK, MIRTK, ITK, simpleITK, 3dSlicer, etc etc etc)

Hope this clarifies this issue.

Kostas
